test-runner/src/main.rs

```diff
@@ -0,0 +1,220 @@
+            .arg(&config.registry_subgraph)
+            .arg("--network-subgraph")
+            .arg(&config.network_subgraph)
+            .arg("--graphcast-network")
+            .arg(&config.graphcast_network)
+            .arg("--topics")
+            .arg(config.topics.join(","))
+            .arg("--coverage")
+            .arg(match config.coverage {
+                CoverageLevel::Minimal => "minimal",
+                CoverageLevel::OnChain => "on-chain",
+                CoverageLevel::Comprehensive => "comprehensive",
+            })
+            .arg("--collect-message-duration")
+            .arg(config.collect_message_duration.to_string())
+            .arg("--waku-log-level")
+            .arg(config.waku_log_level.as_deref().unwrap_or("None"))
+            .arg("--log-level")
+            .arg(&config.log_level)
+            .arg("--slack-token")
+            .arg(config.slack_token.as_deref().unwrap_or("None"))
+            .arg("--slack-channel")
+            .arg(config.slack_channel.as_deref().unwrap_or("None"))
+            .arg("--discord-webhook")
+            .arg(config.discord_webhook.as_deref().unwrap_or("None"))
+            .arg("--persistence-file-path")
+            .arg(config.persistence_file_path.as_deref().unwrap_or("None"))
+            .arg("--log-format")
+            .arg(&config.log_format)
+            .arg("--radio-name")
+            .arg(&config.radio_name)
+            .stdout(Stdio::piped())
+            .spawn()
+            .expect("Failed to start command"),
+    ));
+
+    // Create cleanup struct
+    let _cleanup = Cleanup {
+        sender: Arc::clone(&sender),
+        radio: Arc::clone(&radio),
+    };
+
+    let processes = vec![
+        (Arc::clone(&sender), sender_log_file),
+        (Arc::clone(&radio), radio_log_file),
+    ];
+
+    let mut handlers = Vec::new();
+
+    // Handle the output of each process in a new thread
+    for (process, mut log_file) in processes {
+        let process = Arc::clone(&process);
+
+        let handler = task::spawn(async move {
+            let reader = BufReader::new(
+                process
+                    .lock()
+                    .unwrap()
+                    .stdout
+                    .take()
+                    .expect("Failed to capture stdout"),
+            );
+
+            // Read the stdout line by line and write to the corresponding log file
+            for line in reader.lines() {
+                match line {
+                    Ok(line) => {
+                        writeln!(log_file, "{}", line).expect("Failed to write to log file");
+                    }
+                    Err(err) => eprintln!("Error: {}", err),
+                }
+            }
+        });
+
+        handlers.push(handler);
+    }
+
+    // Wait for 2 minutes asynchronously
+    sleep(Duration::from_secs(120)).await;
+
+    // Read the content of the state.json file
+    let state_file_path = "./test-runner/state.json";
```

Comment:
It would be great if the tests themselves can be separated out from test-runner main, such that test-runner can call different sets of tests. just an idea to make test process more modular and streamlined when more tests are added